### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by EdLichtman**
## Description
Fixes some misleading documentation (ParameterType.GetOrPost was suggesting that it was a Cookie Parameter, and UrlSegmentParameter suggested it modifies the Query String) and adds additional documentation (adds information to the ResponseStatus to explain what each Response Status means). 

Also, adds tests to ensure UrlSegment operates properly and as an enhanced form of documentation for users looking into this code.

## Purpose
This pull request is a:

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/restsharp/RestSharp?pullRequest=2002) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/restsharp/RestSharp?pullRequest=2002) before we can accept your contribution.<br/><br/>:x: EdLichtman [sign now](https://cla.dotnetfoundation.org/restsharp/RestSharp?pullRequest=2002)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/restsharp/RestSharp?pullRequest=2002) it.</sub>

File: src/RestSharp/Enum.cs [hunk 1]
```diff
@@ -19,9 +20,44 @@ namespace RestSharp;
...
+    /// A <see cref="Parameter"/> that will added to the QueryString for GET, DELETE, OPTIONS and HEAD requests; and form for POST and PUT requests.
     /// </summary>
-    GetOrPost, UrlSegment, HttpHeader, RequestBody, QueryString
+    /// <remarks>
+    /// See <see cref="GetOrPostParameter"/>.
+    /// </remarks>
+    GetOrPost,
+
+    /// <summary>
+    /// A <see cref="Parameter"/> that will be added to part of the url by replacing a {placeholder} within the absolute path.
```
Comment (tacosontitan) [inline src/RestSharp/Enum.cs:31]: Utilizing the `c` tag for `{placeholder}` will allow documentation generators to use code tags:
```csharp
/// A <see cref="Parameter"/> that will be added to part of the url by replacing a <c>{placeholder}</c> within the absolute path.
```

**Review by tacosontitan — APPROVED** [review]
Overall, this looks like a solid contribution to the documentation for the library.

File: src/RestSharp/Enum.cs (hunk 1 above)
Comment (EdLichtman) [reply src/RestSharp/Enum.cs:31]: What exactly would this do?

Comment (alexeyzimarev) [reply src/RestSharp/Enum.cs:31]: It's the `<code>` tag that would change the typeface when the code inside the tag is rendered in the popup.

**Comment by alexeyzimarev** [comment]
Would you be able to make this PR targeting the `v109` branch instead of `dev`?

**Comment by EdLichtman** [comment]
Hrm... It seems I may need to pull latest v109, otherwise I may be including commits and changes that you guys aren't ready to put into v109.

**Review by tacosontitan — COMMENTED** [review]
REDACTED; reviewed prior to refreshing and seeing base change and closure.

**Pull request closed without merging**